### PR TITLE
Downstream command protocol example.

### PR DESCRIPTION
**Pull request by peng-yongsheng**
Please answer these questions before submitting pull request

- Why submit this pull request?
- [ ] Bug fix
- [X] New feature provided
- [ ] Improve performance

Example command:
1. Meta data reset: If command specified service id or instance UUID, agent need to check that the conditions are met. Wait seconds represent agent need to stop sending trace to backend until timeout.
2. Trace ignore: Add or delete ignore path config to agent ignore plugin.

### CODE REVIEW

File: apm-protocol/apm-network/src/main/java/org/apache/skywalking/apm/network/trace/component/command/MetaDataResetCommand.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+ * @author peng-yongsheng
+ */
+public class MetaDataResetCommand extends BaseCommand implements Serializable {
+
+    private KeyStringValuePair.Builder waitArguments = KeyStringValuePair.newBuilder();
+    private KeyStringValuePair.Builder specifiedArguments = KeyStringValuePair.newBuilder();
+
+    public MetaDataResetCommand() {
+        super("MetaDataReset");
+        waitArguments.setKey("Wait_Seconds");
```
Comment (wu-sheng) [inline apm-protocol/apm-network/src/main/java/org/apache/skywalking/apm/network/trace/component/command/MetaDataResetCommand.java]: Could we keep all lower case?

File: apm-protocol/apm-network/src/main/java/org/apache/skywalking/apm/network/trace/component/command/MetaDataResetCommand.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+/**
+ * @author peng-yongsheng
+ */
+public class MetaDataResetCommand extends BaseCommand implements Serializable {
+
+    private KeyStringValuePair.Builder waitArguments = KeyStringValuePair.newBuilder();
+    private KeyStringValuePair.Builder specifiedArguments = KeyStringValuePair.newBuilder();
+
+    public MetaDataResetCommand() {
+        super("MetaDataReset");
```
Comment (wu-sheng) [inline apm-protocol/apm-network/src/main/java/org/apache/skywalking/apm/network/trace/component/command/MetaDataResetCommand.java]: Metadata is a word. We should use MatadataSet.

File: apm-protocol/apm-network/src/main/java/org/apache/skywalking/apm/network/trace/component/command/MetaDataResetCommand.java (hunk 1 above)
Comment (wu-sheng) [reply apm-protocol/apm-network/src/main/java/org/apache/skywalking/apm/network/trace/component/command/MetaDataResetCommand.java]: Or WaitSeconds? I think consistent style helps further review.

Comment (peng-yongsheng) [reply apm-protocol/apm-network/src/main/java/org/apache/skywalking/apm/network/trace/component/command/MetaDataResetCommand.java]: Fixed

File: apm-protocol/apm-network/src/main/java/org/apache/skywalking/apm/network/trace/component/command/MetaDataResetCommand.java (hunk 2 above)
Comment (peng-yongsheng) [reply apm-protocol/apm-network/src/main/java/org/apache/skywalking/apm/network/trace/component/command/MetaDataResetCommand.java]: Fixed

File: apm-protocol/apm-network/src/main/java/org/apache/skywalking/apm/network/trace/component/command/MetadataResetCommand.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    public void specifiedService(int serviceId) {
+        specifiedArguments.setKey("SpecifiedService");
+        specifiedArguments.setValue(String.valueOf(serviceId));
+    }
+
+    public void specifiedInstance(String instanceUUID) {
+        specifiedArguments.setKey("SpecifiedInstance");
+        specifiedArguments.setValue(instanceUUID);
```
Comment (wu-sheng) [inline apm-protocol/apm-network/src/main/java/org/apache/skywalking/apm/network/trace/component/command/MetadataResetCommand.java]: Is this really necessary? Would you support downstream different serviceId and serviceInstanceId to agent? At least in language agents, I am not sure.

**Comment by wu-sheng** [comment]
cc @apache/skywalking-committers

**Pull request merged**